client/state/data-layer/wpcom/sites/activity/from-api.js

```diff
@@ -63,6 +61,7 @@ export function processItemBase( item ) {
 		activityStatus: get( item, 'status' ),
 		activityTitle: get( item, 'summary', '' ),
 		activityTs: Date.parse( published ),
+		activityTargetTs: get( item, 'object.target_ts', null ),
```

Comment:
The separate functions were intended to process different "pieces" independently, especially those that may vary depending on the type of Activity like `object` or `target`.

Just clarifying intent, no objection if you choose to take a different approach.